### PR TITLE
Re-implements command line interface

### PR DESCRIPTION
**Pull request by willu47**
(Almost) [Finishes #151515037](https://www.pivotaltracker.com/story/show/151515037)

### New additions and updates
* `smif run` now runs the first model run in the list (hard coded)
* I've added various additional methods to the data layer (but not yet to the interface)
* The fixture project in `tests/fixtures/single_run/` is now running and all in the correct new format and folder structure
* @RoaldL - I've hacked at some of your code in the DatafileInterface. I've made some pythonic stylistic changes here and there.  See comments on 0feb3ad

### Next steps
* build the cli dynamically so that the command `smif` shows a list of available model runs and `smif run <model_run_id>` runs that model run
* load parameter values from the narratives

### Outstanding
* tests still fail.  I'd like to get these working so we can merge this develop branch into master for v0.5.0 before the mini-assembly if possible...
  - @tomalrussell - there's one issue with the fixture models and the `converge` method in `ModelSet` *FIXED* - datafileinterface read in convergence values as strings, which were passed to model set arguments
* all validation has been removed - will the GUI do this, or should we re-instate this @tomalrussell & @RoaldL? *FIXED* - validation removed from CLI as will be placed in GUI (validation module still in cli package for use by GUI server)

### CODE REVIEW

File: smif/data_layer/__init__.py [hunk 1]
```diff
@@ -480,25 +547,33 @@ def read_scenario_data(self, scenario_name):
...
+                    # Read the scenario data from file
+                    filepath = os.path.join(self.file_dir['scenarios'], filename)
+                    data[param['name']] = self._get_data_from_csv(filepath)
 
-        # Read the scenario data from file
-        filepath = os.path.join(self.file_dir['scenarios'], filename)
-        reader = csv.DictReader(open(filepath))
+        return data
 
+    def _get_data_from_csv(self, filepath):
```
Comment (RoaldL) [inline smif/data_layer/__init__.py]: Would be nicer to move this down with the other 'private' methods

File: smif/data_layer/__init__.py [hunk 2]
```diff
@@ -470,6 +519,24 @@ def read_scenario_set(self, scenario_set_name):
 
         return filtered_scenario_data
 
+    def read_scenario_config(self, scenario_name):
```
Comment (RoaldL) [inline smif/data_layer/__init__.py]: The name is not consistent with the other methods
I would either call this 'read_scenario' or update the existing functions to

read_scenario_sets(self) -> read_scenario_config_sets(self)
read_scenario_set(self, scenario_set_name) -> read_scenario_config_set(self, scenario_config_set_name)
write_scenario_set(self, scenario_set): -> ..
update_scenario_set(self, scenario_set): -> ..
write_scenario(self, scenario): -> write_scenario_config(self, scenario):
update_scenario(self, scenario): -> ..

Comment (RoaldL) [reply smif/data_layer/__init__.py]: In fact I would rather use definition instead as config, to stay in a similar style with region_definition and interval_definition

File: smif/cli/__init__.py [hunk 3]
```diff
@@ -196,20 +198,91 @@ def run_model(args):
     """Runs the model specified in the args.model argument
```
[the remark this answers is not in the document]
Comment (willu47) [reply smif/cli/__init__.py]: 👍

**Pull request merged**